tasks/RTPTask.cpp

```diff
@@ -0,0 +1,146 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include <base-logging/Logging.hpp>
+#include <gst/gstcaps.h>
+#include <set>
+
+#include "Helpers.hpp"
+#include "RTPTask.hpp"
+
+using namespace std;
+using namespace gstreamer;
+using namespace base::samples::frame;
+
+RTPTask::RTPTask(std::string const& name)
+    : RTPTaskBase(name)
+{
+}
+
+RTPTask::~RTPTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See Task.hpp for more detailed
+// documentation about them.
+
+void RTPTask::extractPipelineStatus()
+{
+    GstStructure* stats = NULL;
+    g_object_get(m_rtpbin, "stats", &stats, NULL);
```

Comment:
I don't think `g_object_get` is the one responsible for GstStructure memmory allocation https://docs.gtk.org/gobject/method.Object.get.html
```suggestion
    GstStructure stats;
    g_object_get(m_rtpbin, "stats", &stats, NULL);
```